### PR TITLE
Apply Rector fixes, add PHPStan baseline

### PR DESCRIPTION
**Pull request by sylfabre**
## Summary
- Made data provider methods static (Rector)
- Added PHPStan baseline for `denormalize()` return type covariance (Symfony interface edge case)

## Test plan
- [x] Rector clean
- [x] PHPStan 0 errors (1 baselined)
- [x] PHPUnit passes (15 tests, 22 assertions)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -4,4 +4,5 @@ parameters:
         - tests/
 
 includes:
-    - vendor/assoconnect/php-quality-config/phpstan.extension.neon
\ No newline at end of file
+    - vendor/assoconnect/php-quality-config/phpstan.extension.neon
+    - phpstan-baseline.neon
```
Comment (sylfabre) [inline phpstan.neon.dist]: Missing blank line at the end
